### PR TITLE
MCR-2789 Added property MCR.Solr.Disallowed.Facets to disallow certain facets

### PR DESCRIPTION
**Pull request by Possommi**
…n facets

[Link to jira](https://mycore.atlassian.net/browse/MCR-2789).

### CODE REVIEW

**Review by fluetze — APPROVED** [review]
It is good to have this feature, although it is incomplete:
One could also use the json facet api to get faceted output,
https://solr.apache.org/guide/8_1/json-facet-api.html

File: mycore-solr/src/main/java/org/mycore/solr/proxy/MCRSolrProxyServlet.java [hunk 1]
```diff
@@ -275,6 +276,15 @@ private void handleQuery(String queryHandlerPath, HttpServletRequest request, Ht
         solrHttpMethod.releaseConnection();
     }
 
+    private void filterParams(ModifiableSolrParams solrParameter) {
+        MCRConfiguration2.getString("MCR.Solr.Disallowed.Facets").ifPresent(df -> {
+            String[] disallowedFacets = df.split(" *, *");
```
Comment (rsteph-de) [inline mycore-solr/src/main/java/org/mycore/solr/proxy/MCRSolrProxyServlet.java]: The parameter of split looks unusual - I would prefer: `split("\\s*,\\s*")`.
We have also `MCRConfiguration2::splitValue` to process a comma-separated list of property-values.
There are lots of examples on how to use it in our code.

File: mycore-solr/src/main/java/org/mycore/solr/proxy/MCRSolrProxyServlet.java [hunk 2]
```diff
@@ -275,6 +276,14 @@ private void handleQuery(String queryHandlerPath, HttpServletRequest request, Ht
         solrHttpMethod.releaseConnection();
     }
 
+    private void filterParams(ModifiableSolrParams solrParameter) {
```
Comment (rsteph-de) [inline mycore-solr/src/main/java/org/mycore/solr/proxy/MCRSolrProxyServlet.java:279]: I think this is better readable:
```
MCRConfiguration2.getString("MCR.Solr.Disallowed.Facets")
            .map(MCRConfiguration2::splitValue).orElse(Stream.empty())
            .forEach(disallowedFacet -> {
                solrParameter.remove("facet.field", disallowedFacet);
            });
```

**Review by rsteph-de — DISMISSED** [review]
see comment

File: mycore-solr/src/main/java/org/mycore/solr/proxy/MCRSolrProxyServlet.java (hunk 2 above)
Comment (Possommi) [reply mycore-solr/src/main/java/org/mycore/solr/proxy/MCRSolrProxyServlet.java:279]: To be honest I think this just a matter of taste.

**Review by rsteph-de — APPROVED** [review]

**Pull request merged**